src/signed-data-store/signed-data-store.ts

```diff
@@ -0,0 +1,114 @@
+import { clearInterval } from 'timers';
+import { BigNumber, ethers } from 'ethers';
+import { LocalSignedData, SignedData, AirnodeAddress, DataStore, TemplateId } from '../types';
+import { logger } from '../logger';
+
+// A simple in-memory data store implementation - the interface allows for swapping in a remote key/value store
+
+let signedApiStore: Record<AirnodeAddress, Record<TemplateId, LocalSignedData>> = {};
+let pruner: NodeJS.Timeout | undefined;
+
+export const checkMessage = ({ airnode, templateId, timestamp, signature, encodedValue }: SignedData) => {
+  // 'data' is: ethers.utils.defaultAbiCoder.encode(['int256'], [beaconValue]);
+
+  const message = ethers.utils.arrayify(
+    ethers.utils.solidityKeccak256(['bytes32', 'uint256', 'bytes'], [templateId, timestamp, encodedValue])
+  );
+
+  const signerAddr = ethers.utils.verifyMessage(message, signature);
+  return signerAddr !== airnode;
+};
+
+export const checkSignedDataIntegrity = (signedData: SignedData) => {
+  const { airnode, templateId, timestamp, encodedValue } = signedData;
+
+  if (parseInt(timestamp) * 1_000 > Date.now()) {
```

Comment:
I don't think we need this. We trust the pusher (data provider) and if they are malicious they can straight up misreport (but their signature is there so they won't do it). 

I think we can just have the `verifySignedData`

Reply:
Just noting that we decided in the meeting to remove this 👌

Reply:
Update:
- Mirror what's in the contract
- Log a WARN message if timestamp is in the future.

Reply:
Yeah, I take it that you will do it in this repo and will on only create an issue in signed-api repo.

Reply:
Looks like you've already created the issue, thanks :) https://github.com/api3dao/signed-api/issues/85